homeassistant/components/stream/__init__.py

```diff
@@ -73,7 +88,7 @@ def request_stream(hass, stream_source, *, fmt="hls", keepalive=False, options=N
         if not stream.access_token:
             stream.access_token = secrets.token_hex()
             stream.start()
-        return hass.data[DOMAIN][ATTR_ENDPOINTS][fmt].format(stream.access_token)
+        return hass.data[DOMAIN][ATTR_ENDPOINTS][fmt][0].format(stream.access_token)
```

Comment:
Instead of adding the logic above, I would just check the `master_playlist` boolean here and return `0` or `1` depending on the value, keeping all other logic the same.